Database/Entities/PlayerHighestRanks.cs

```diff
@@ -0,0 +1,55 @@
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Diagnostics.CodeAnalysis;
+using Database.Enums;
+
+namespace Database.Entities;
+
+/// <summary>
+/// Represents the highest ranks of a player for a given ruleset
+/// </summary>
+[Table("player_highest_ranks")]
+[SuppressMessage("ReSharper", "CollectionNeverUpdated.Global")]
+[SuppressMessage("ReSharper", "PropertyCanBeMadeInitOnly.Global")]
+public class PlayerHighestRanks : UpdateableEntityBase
+{
+    /// <summary>
+    /// The ruleset in which this highest rank record occurred in
+    /// </summary>
+    [Column("ruleset")]
+    public Ruleset Ruleset { get; set; }
+
+    /// <summary>
+    /// The highest global rank for this user in the given ruleset
```

Comment:
You can probably remove 'in the given ruleset' from all docstrings here. Its implied by the docstring on the class itself.